src/main/java/mate/academy/web/filter/AuthorizationFilter.java

```diff
@@ -0,0 +1,74 @@
+package mate.academy.web.filter;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.academy.lb.Injector;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+
+public class AuthorizationFilter implements Filter {
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private static final String USER_ID = "user_id";
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+    private Map<String, List<Role.RoleName>> protectedUrls = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls.put("/users/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/all/admin", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/all/admin", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/all", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-carts/products/add", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/products/get", List.of(Role.RoleName.USER));
```

Comment:
you also need to add
for ADMIN:

DeleteUserController
DeleteOrderController
DeleteProductController

for USER:

RemoveProductFromCartController
CompleteOrderController